App/Redux/MigrationRedux.ts

```diff
@@ -79,6 +104,25 @@ export function reducer(state: MigrationState = initialState, action: MigrationA
       const { photosCount, threadsCount } = action.payload
       return { ...state, photosCount, threadsCount }
     }
+    case getType(actions.announceMigration): {
+      const announcement: PeerDetails = action.payload
+      return { ...state, announcement }
+    }
+    case getType(actions.announceSuccess): {
+      return { ...state, announcement: undefined }
```

Comment:
Still need to get to the saga, but first thought here is that `announceMigration` probably doesn't need to modify the state at all. It will just trigger a saga that does a network request. I'd imagine `accounceSuccess` will be when we want to store some value that represents the fact that we've previously successfully announced the migration.

Reply:
Ok i see how you're doing this now... using the absence of `accounceMigration` as the success state. Cool.